tests/testw3emc.f

```diff
@@ -0,0 +1,317 @@
+          TOTAL = 0
+          DO N=1,NNUM
+            WRITE (LABEL(2:5),FMT='(I4.4)') N
+            LABEL(1:1) = CHAR(78)
+            CALL GBYTEC(CBUF,LGRIB,8*NS+8*20,8*4)
+            TOTAL = TOTAL + LGRIB
+            CIS='GRIB'//CBUF(NS+22:NS+25)
+            CPDS=CBUF(NS+26:NS+53)
+            KPDS=CBUF(NS+113:NS+172)
+            CALL W3FP11(CIS,CPDS,CTITLE,IRET)
+            IF (IRET.NE.0) CTITLE=' UNKNOWN FIELD'
+            WRITE (MPDS(1:64),FMT='(4(8Z2.2))') (CPDS(J:J),J=1,28)
+            NPDS(1:8)   = MPDS(1:8)
+            NPDS(9:9)   = ' '
+            NPDS(10:17) = MPDS(9:16)
+            NPDS(18:18) = ' '
+            NPDS(19:26) = MPDS(17:24)
+            NPDS(27:27) = ' '
+            NPDS(28:35) = MPDS(25:32)
+            NPDS(36:36) = ' '
+            NPDS(37:44) = MPDS(33:40)
+            NPDS(45:45) = ' '
+            NPDS(46:53) = MPDS(41:48)
+            NPDS(54:54) = ' '
+            NPDS(55:62) = MPDS(49:56)
+            NPDS(63:63) = ' '
+C
+            IF ( NPDS(57:58) .EQ. '02' ) THEN
+               WRITE(MPDS(57:120),FMT='(3(8Z2.2))')(KPDS(J:J),J=1,24)
+               EPDS(1:8)   = MPDS(57:64)
+               EPDS(9:9)   = ' '
+               EPDS(10:17) = MPDS(65:72)
+               EPDS(18:18) = ' '
+               IF ((NPDS(19:20).EQ.'BF'.OR.NPDS(19:20).EQ.'C0')
+     &           .OR.(EPDS(3:4).EQ.'04'.OR.EPDS(3:4).EQ.'05')) THEN
+                  EPDS(19:26) = MPDS(73:80)
+                  EPDS(27:27) = ' '
+                  EPDS(28:35) = MPDS(81:88)
+                  EPDS(36:36) = ' '
+                  EPDS(37:44) = MPDS(89:96)
+                  EPDS(45:45) = ' '
+                  EPDS(46:53) = MPDS(97:104)
+                  WRITE(OUTPUT,2800)LABEL(1:5),NPDS,EPDS,CTITLE(2:26),
+     &            CTITLE(28:45),CTITLE(46:47),CTITLE(51:62),LGRIB
+               ELSE
+                  WRITE(OUTPUT,2801)LABEL(1:5),NPDS,EPDS,CTITLE(2:26),
+     &            CTITLE(28:45),CTITLE(46:47),CTITLE(51:62),LGRIB
+               END IF
+            ELSE
+               WRITE(OUTPUT,2802)LABEL(1:5),NPDS,CTITLE(2:26),
+     &         CTITLE(28:45),CTITLE(46:47),CTITLE(51:62),LGRIB
+            ENDIF
+            NS=NS+NLEN
+          ENDDO
+      ELSE
+          PRINT *,' ... COULD NOT READ INDEX FILE = ',CGI(1:NCGI)
+      ENDIF
+      WRITE (OUTPUT,2400) CGI(1:NCGI), NNUM, IARG, TOTAL
+      CALL BACLOSE(32,IRET)
+C
+      CALL W3UTCDAT (ITIME)
+      JREW    = 0
+      JGDS    = -1
+      MPDST   = -1
+      CPARM = 'KWBC'
+      CALL W3AS00(NPARM,CPARM,IER)
+      CPARM = 'KWBC'
+c
+      CALL GETGBP(LUGB,LUGI,MXSIZ3,JREW,MPDST,JGDS,
+     &      itot,KREW,KPDST,KGDS,GRIB,IRET)
+      print *, ' '
+      IF (CHKIRET.EQ.0) THEN
+          print *,'Testing NCEPLIBS-w3emc: SUCCESS !'
+      ELSE
+          print *,'Testing NCEPLIBS-w3emc: FAILED !'
+      ENDIF
+      print *, ' '
+c
+      END
+C-----------------------------------------------------------------------
+      SUBROUTINE WRGI1H(LUGI,NLEN,NNUM,CGB)
```

Comment:
Was this routine copied from a different library? We should try to avoid copy-pasting code and use the library directly, if possible.

Reply:
This routine is from W3EMC. You can remove the hostnam or leave it as blank. The CHEAD is header of a grib index.

Reply:
It uses to write header for an index file.  You can not create an index without it.